src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/components/options_list_popover_action_bar.tsx

```diff
@@ -69,6 +77,37 @@ export const OptionsListPopoverActionBar = ({
     [searchTechnique, compatibleSearchTechniques]
   );
 
+  const loadMoreOptions = useCallback(async (): Promise<OptionsListSuggestions | undefined> => {
+    componentApi.setRequestSize(Math.min(totalCardinality, MAX_OPTIONS_LIST_REQUEST_SIZE));
+    componentApi.loadMoreSubject.next(); // trigger refetch with loadMoreSubject
+    return lastValueFrom(componentApi.availableOptions$.pipe(take(2)));
+  }, [componentApi, totalCardinality]);
+
+  const hasNoOptions = availableOptions.length < 1 || totalCardinality < 1;
+  const hasTooManyOptions = showOnlySelected
+    ? selectedOptions.length > MAX_OPTIONS_LIST_BULK_SELECT_SIZE
+    : totalCardinality > MAX_OPTIONS_LIST_BULK_SELECT_SIZE;
+
+  const isBulkSelectDisabled = hasNoOptions || hasTooManyOptions;
+
+  const handleBulkAction = useCallback(
+    async (bulkAction: (keys: string[]) => void) => {
+      if (showOnlySelected) {
+        bulkAction(selectedOptions as string[]);
+        return;
+      }
+
+      if (totalCardinality > availableOptions.length) {
+        const newAvailableOptions = (await loadMoreOptions()) ?? [];
+        bulkAction(newAvailableOptions.map(({ value }) => value as string));
+        return;
+      }
```

Comment:
I wonder if we need a more obvious loading state for this?

![May-27-2025 15-26-12](https://github.com/user-attachments/assets/4b9c6d81-537f-4292-a9f2-5f1a9d24df00)

Reply:
Would it be sufficient to just disable select/deselect buttons while the control is loading? I don't think we need any additional loading indicators since we have the spinner in the base control and the loading bar at the bottom of the popover.

Reply:
> m pretty sure that loading indicator is just the Eui icon being async imported

I know! I just don't love how we have two consecutive loading states happening, so if we could simulate the same "loading" state, I think this would help with the feeling of "I clicked this button but nothing is happening!"